### ISSUE TITLE
Commands failing

### ISSUE THREAD

**Issue by artola** [biomejs/biome#4225]
### Environment information

```block
CLI:
  Version:                      1.9.3
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.13.1"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "yarn/4.5.0"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 false

Workspace:
  Open Documents:               0
```

### What happened?

Running: `biome format --write` fixes "n" files, I do git reset the changes, and re-run the command without file changes this time. Only after re-saving such files, the tool runs the command properly.

### Expected result

The command should be idempotent. If not possible or intended due performance, then a flag should allow to (for example: `--force`, `--no-cache`).

Also trying `yarn biome format --write --since HEAD` throws an error: 

```
  ✖ Incompatible arguments since and changed
```

That makes no sense to me. I was just looking a workaround to re-run.

Sorry if I am wrong, but I did not find in the docs other instructions.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by artola** [comment]
@dyc3 Here the [repro](https://github.com/artola/biome-issue-4225). It is easy to test in any project, basically:

```sh
yarn format --write --since $(git rev-parse HEAD)
```

**Comment by dyc3** [comment]
I think there's multiple problems here:

- `--since` is only valid if `--changed` is also used, and the error message we emit for when you forget `--changed` is not helpful, which is a bug
- when biome checks to see what files have changed when you do `--changed --since HEAD`, its comparing the difference from HEAD to HEAD, and sees that no files changed.

I will admit that its confusing behavior for `--since HEAD` to not pick up dirty files in your work tree. We should do that.

I believe `--staged` will fit your use case as a workaround, as long as you stage the files first.

**Comment by ematipico** [comment]
Closing. The issues has been fixed in one of latest versions of Biome.
